example_sns_init.yaml

```diff
@@ -25,96 +31,338 @@ description: >
+    #
+    # there be for a particular action given a particular `RewardRate`
+    # configuration. How exactly does inflation project to voting rewards?
+    #
+    # The voting reward rate controls how quickly the supply of the SNS token
+    # increases. For example, setting `initial` to `2%` will cause the supply to
+    # increase by at most `2%` per year. A higher voting reward rate
+    # incentivizes users to participate in governance, but also results in
+    # higher inflation.
+    #
+    # The initial and final reward rates can be set to have a higher reward rate
+    # at the launch of the SNS and a lower rate further into the SNS’s lifetime.
+    # The reward rate falls quadratically from the `initial` rate to the `final`
+    # rate over the course of `transition_duration`.
+    #
+    # Setting both `initial` and `final` to `0%` will result in the system not
+    # distributing voting rewards at all.
     RewardRate:
+        # The initial reward rate that SNS voting rewards will increase the SNS
+        # token supply per year.
         initial: 10%
+
+        # The final reward rate that SNS voting rewards will increase the SNS
+        # token supply per year. This rate is reached after `transition_duration`
+        # and remains at this level unless changed by an SNS proposal.
+        #
+        # TODO[ATG]: is there a requirement that `final` <= `initial`?
         final: 2.25%
+
+        # The voting reward rate falls quadratically from `initial` to `final`
+        # over the time period defined by `transition_duration`.
+        #
+        # Values of 0 result in the reward rate always being `final`.
         transition_duration: 12 years
 
+# Configuration of the initial token distribution of the SNS. You can configure
+# how SNS tokens are distributed in each of the three groups:
+# (1) tokens that are given to the original developers of the dapp,
+# (2) treasury tokens that are owned by the SNS governance canister, and
+# (3) tokens which are distributed to the decentralization swap participants.
+#
+# The initial token distribution must satisfy the following preconditions to be
+# valid:
+#    - The sum of all developer tokens in E8s must be less than `u64::MAX`.
+#    - The Swap's initial balance (see group (3) above) must be greater than 0.
+#    - The Swap's initial balance (see group (3) above) must be greater than or
+#      equal to the sum of all developer tokens.
 Distribution:
-
+    # The neurons distributed at genesis to the development team of the dapp
+    # being decentralized. Each element in this list specifies one such neuron,
+    # including its stake, controlling principal, unique memo identifying this
+    # neuron, dissolve delay, and a vesting period. Even though these neurons
+    # are distributed at genesis, they are locked in a (restricted)
+    # pre-initialization mode until the decentralization swap is completed.
     Neurons:
```

Comment:
IIRC you cannot have multiple dev neurons with the same controller and the same memo as this would give a clash in ledger accounts. Is this the case or are there other conditions? 
If so, would it maybe be helpful to say so?

Reply:
I think it would imply the property that I am looking for but I would be more precise (it is a memo per controller) and more explicit:
How about "unique memo identifying this neuron (for each controller, every neuron needs a unique memo)"

Reply:
Controller is mostly used in this file to refer to dapp controllers, not principal controllers.

So I came up with this:

```
memo identifying this neuron (every neuron that a user has must be identified by a unique memo)
```

Wdyt?